docs/user/metrics/custom_distribution.md

```diff
@@ -0,0 +1,86 @@
+# Custom Distribution
+
+Custom distributions are used to record the distribution of arbitrary values.
+
+It should be used only when direct control over how the histogram buckets are computed is required.
+Otherwise, look at the standard distribution metric types:
+
+* [Timing Distributions](timing_distribution.md))
+
+> Note: Custom distributions are currently only allowed for GeckoView metrics (the `gecko_datapoint` parameter is present).
+
+## Configuration
+
+Custom distributions have the following required parameters:
+
+  - `range_min`: The minimum value of the first bucket
+  - `range_max`: The minimum value of the last bucket
+  - `bucket_count`: The number of buckets
+  - `histogram_type`: 
+    - `linear`: The buckets are evenly spaced
+    - `exponential`: The buckets follow a natural logarithmic distribution
+
+In addition, the metric should specify:
+
+  - `unit`: The unit of the values in the metric. For documentation purposes only -- does not affect data collection.
+
+If you wanted to create a custom distribution to the fraction of memory that is overhead as a percentage, first you need to add an entry for it to the `metrics.yaml` file:
+
+```YAML
+memory:
```

Comment:
minor-nit: is it worth using a different example here? We're adding memory distributions as well and this might be a bit confusing to people (even though the unit is clearly different!). I'd suggest something that has nothing to do with memory (e.g. pixel distribution?)

Reply:
Good point.  I wanted to use a "real" example, but I agree a non-memory one might be less confusing.

Reply:
It seems all the other "real" ones are timing adjacent and will have a similar confusion with timing.  I guess I'll make one up about pixels.